### PR TITLE
HashCube game's specific changes

### PR DESCRIPTION
**Pull request by rampr**

### CODE REVIEW

**Review by shijasmhd — APPROVED** [review]

**Review by thesayraj — APPROVED** [review]

File: cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java [hunk 1]
```diff
@@ -172,4 +173,10 @@ public void setWebViewRect(int left, int top, int maxWidth, int maxHeight) {
         layoutParams.gravity = Gravity.TOP | Gravity.LEFT;
         this.setLayoutParams(layoutParams);
     }
+
+
+    @Override
+    public boolean onKeyDown(int keyCode, KeyEvent event) {
+        return true;
```
Comment (hatim-developer) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java:195]: you are missing some code here @rampr 

https://bitbucket.org/hashcube/frameworks/commits/44b52b199107077e22fe03375e9cf4eea1860707

File: cocos/storage/local-storage/LocalStorage.cpp [hunk 2]
```diff
@@ -33,19 +33,26 @@
 
 #if (CC_TARGET_PLATFORM != CC_PLATFORM_ANDROID)
 
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_IOS)
+    // Comment below line to disable User Deafualts Usage for Local Storage
+    #define USE_USER_DEFAULT_LS
```
Comment (hatim-developer) [inline cocos/storage/local-storage/LocalStorage.cpp:38]: is this right @rampr ? we are not using user default local storage?

**Review by anil-kumar07 — APPROVED** [review]

**Review by hatim-developer — CHANGES_REQUESTED** [review]

File: cocos/storage/local-storage/LocalStorage.cpp (hunk 2 above)
Comment (rampr) [reply cocos/storage/local-storage/LocalStorage.cpp:38]: Let me check

File: cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java (hunk 1 above)
Comment (rampr) [reply cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java:195]: Thanks @hatim-developer . Will check and update

Comment (rampr) [reply cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java:195]: This has been missed, will update the changes

File: cocos/storage/local-storage/LocalStorage.cpp (hunk 2 above)
Comment (rampr) [reply cocos/storage/local-storage/LocalStorage.cpp:38]: I have to check, this has been copied over from our version of cocos2d. If this is not used, this is dead code

File: cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java (hunk 1 above)
Comment (rampr) [reply cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxWebView.java:195]: Updated

File: cocos/storage/local-storage/LocalStorage.cpp (hunk 2 above)
Comment (rampr) [reply cocos/storage/local-storage/LocalStorage.cpp:38]: @hatim-developer  this is the behaviour for ios and is needed

**Review by hatim-developer — APPROVED** [review]

**Pull request merged**